### DISCUSSION TITLE
GITHUB_TOKEN

### DISCUSSION THREAD

**Discussion by stephanepx** [obytes/react-native-template-obytes#364] [Q&A] [answered]
Hi guys!

I was wondering how some workflow files can use GITHUB_TOKEN in secrets if we don't set it. 

For example in lint-ts.yml file, line 39, we have this : `github_token: ${{ secrets.GITHUB_TOKEN }}`
And in line 11 we have this comment : "🚨 GITHUB SECRETS REQUIRED: NONE"

I don't understand how this workflow will access this secret.

Thanks!

**Comment by yjose** [comment, marked as answer]
Hey @stephanepx, normally GitHub automatically creates a unique GITHUB_TOKEN secret for each workflow. This token works as expected for most workflows, but sometimes the permissions are insufficient for writing and tag creation. That's why, for example, we require a custom GitHub token for [app release ](https://github.com/obytes/react-native-template-obytes/blob/5e4d8de4bf82d26c2c3efe534cae0b8e5774a395/.github/workflows/new-app-version.yml#L44)

Any way you can read more [here](https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication)

**Reply by stephanepx** [reply]
Thanks a lot !
